### PR TITLE
Added 'Blessed' icons for core plugins

### PR DESCRIPTION
**Pull request by hrithiksawhney**
## Hey, I just made a Pull Request!

In this Pull Request, I have selected a few icons from v4 MUI Icons set to represent major parts of Backstage, for use in buttons and links into these components: 
catalog: MenuBook Icon
scaffolder & templates: CreateNewFolder Icon
techdocs: Subject Icon
search: Search Icon

I have added these icons to the list.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by freben** [comment]
Alright, two more things.

You'll need to add a changeset, and add a DCO signature onto your commit. See the details of the failed build entry here for details, and some notes in CONTRIBUTING.md

**Comment by hrithiksawhney** [comment]
> Alright, two more things.
> 
> You'll need to add a changeset, and add a DCO signature onto your commit. See the details of the failed build entry here for details, and some notes in CONTRIBUTING.md

Hi, I have added the changeset and also added a DCO signature onto my commit.

File: .changeset/spicy-panthers-thank.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+'@backstage/core-app-api': major
```
Comment (freben) [inline .changeset/spicy-panthers-thank.md:2]: Not a major change - leave this as `patch`

Comment (hrithiksawhney) [reply .changeset/spicy-panthers-thank.md]: Done

**Review by freben — APPROVED** [review]
Alright!

**Comment by freben** [comment]
Just have to wait for it to turn green and then I'll merge some time later today

**Comment by freben** [comment]
You'll notice it errors on unknown words in the changeset. You should probably just simplify it to say "Added default icons for the scaffolder, techdocs, and search."

**Comment by hrithiksawhney** [comment]
> You'll notice it errors on unknown words in the changeset. You should probably just simplify it to say "Added default icons for the scaffolder, techdocs, and search."

Yes, I have made changes to changeset.

**Pull request merged**